### PR TITLE
put RestClient and RestResponse behind interfaces

### PR DESCRIPTION
**Pull request by wpeter-sfdc**
So we can write proper unit tests for any services that consume these classes

### CODE REVIEW

**Comment by wpeter-sfdc** [comment]
@kchitalia 
@smcnulty-sfdc

**Comment by jhsfdc** [comment]
:+1:

File: SalesforceSDK/Core/Rest/IRestClient.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
```
Comment (kchitalia) [inline SalesforceSDK/Core/Rest/IRestClient.cs]: Add copyright info here.

**Comment by smcnulty-sfdc** [comment]
:+1:

**Comment by kchitalia** [comment]
:+1:

**Pull request merged**